Tiller Back-off restarting failed container

<!-- If you need help or think you have found a bug, please help us with your issue by entering the following information (otherwise you can delete this text): -->

Output of `helm version`:
```
Client: &version.Version{SemVer:"v2.10.0", GitCommit:"9ad53aac42165a5fadc6c87be0dea6b115f93090", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.10.0", GitCommit:"9ad53aac42165a5fadc6c87be0dea6b115f93090", GitTreeState:"clean"}
```

Output of `kubectl version`:
```
Client Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.2", GitCommit:"bb9ffb1654d4a729bb4cec18ff088eacc153c239", GitTreeState:"clean", BuildDate:"2018-08-07T23:17:28Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.1", GitCommit:"b1b29978270dc22fecc592ac55d903350454310a", GitTreeState:"clean", BuildDate:"2018-07-17T18:43:26Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}
```

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
Rancher

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x50 pc=0x12e7028]
goroutine 70 [running]:
k8s.io/helm/pkg/tiller.(*ReleaseServer).performUpdateForce(0xc4205ca090, 0xc4200c6870, 0x0, 0x0, 0x17b50a0)
/go/src/k8s.io/helm/pkg/tiller/release_update.go:166 +0x188
k8s.io/helm/pkg/tiller.(*ReleaseServer).UpdateRelease(0xc4205ca090, 0x17d5e00, 0xc420737980, 0xc4200c6870, 0xc4205ca090, 0x16b70a5, 0xd)
/go/src/k8s.io/helm/pkg/tiller/release_update.go:43 +0x245
k8s.io/helm/pkg/proto/hapi/services._ReleaseService_UpdateRelease_Handler.func1(0x17d5e00, 0xc420737980, 0x1603f20, 0xc4200c6870, 0x32, 0xc4200c6cd0, 0xc4200c6801, 0x0)
/go/src/k8s.io/helm/pkg/proto/hapi/services/tiller.pb.go:1244 +0x86
k8s.io/helm/vendor/github.com/grpc-ecosystem/go-grpc-prometheus.(*ServerMetrics).UnaryServerInterceptor.func1(0x17d5e00, 0xc420737980, 0x1603f20, 0xc4200c6870, 0xc42084a6e0, 0xc42084a700, 0x48, 0xc42084a580, 0xc420e45a40, 0x4108c8)
/go/src/k8s.io/helm/vendor/github.com/grpc-ecosystem/go-grpc-prometheus/server_metrics.go:86 +0xdb
k8s.io/helm/pkg/tiller.newUnaryInterceptor.func1(0x17d5e00, 0xc420737980, 0x1603f20, 0xc4200c6870, 0xc42084a6e0, 0xc42084a700, 0xc4205f6ab0, 0x4108c8, 0x50, 0x15d6360)
/go/src/k8s.io/helm/pkg/tiller/server.go:60 +0xab
k8s.io/helm/pkg/proto/hapi/services._ReleaseService_UpdateRelease_Handler(0x163dd60, 0xc4205ca090, 0x17d5e00, 0xc420737980, 0xc4200c6780, 0x1712cd8, 0x0, 0x0, 0x43317c, 0xc420001380)
/go/src/k8s.io/helm/pkg/proto/hapi/services/tiller.pb.go:1246 +0x167
k8s.io/helm/vendor/google.golang.org/grpc.(*Server).processUnaryRPC(0xc42023a140, 0x17de900, 0xc42030a580, 0xc420d48000, 0xc4205ca4e0, 0x215fcf0, 0x0, 0x0, 0x0)
/go/src/k8s.io/helm/vendor/google.golang.org/grpc/server.go:826 +0x9c8
k8s.io/helm/vendor/google.golang.org/grpc.(*Server).handleStream(0xc42023a140, 0x17de900, 0xc42030a580, 0xc420d48000, 0x0)
/go/src/k8s.io/helm/vendor/google.golang.org/grpc/server.go:1023 +0x1318
k8s.io/helm/vendor/google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc420044fb0, 0xc42023a140, 0x17de900, 0xc42030a580, 0xc420d48000)
/go/src/k8s.io/helm/vendor/google.golang.org/grpc/server.go:572 +0x9f
created by k8s.io/helm/vendor/google.golang.org/grpc.(*Server).serveStreams.func1
/go/src/k8s.io/helm/vendor/google.golang.org/grpc/server.go:570 +0xa1
```

--- a/pkg/tiller/release_update.go
+++ b/pkg/tiller/release_update.go
@@ -150,6 +150,8 @@ func (s *ReleaseServer) performUpdateForce(req *services.UpdateReleaseRequest) (
 		return nil, err
 	}
 
+	res := &services.UpdateReleaseResponse{}
+
 	newRelease, err := s.prepareRelease(&services.InstallReleaseRequest{
 		Chart:        req.Chart,
 		Values:       req.Values,
@@ -161,11 +163,6 @@ func (s *ReleaseServer) performUpdateForce(req *services.UpdateReleaseRequest) (
 		Timeout:      req.Timeout,
 		Wait:         req.Wait,
 	})
-
-	// update new release with next revision number so as to append to the old release's history
-	newRelease.Version = oldRelease.Version + 1
-
-	res := &services.UpdateReleaseResponse{Release: newRelease}
 	if err != nil {
 		s.Log("failed update prepare step: %s", err)
 		// On dry run, append the manifest contents to a failed release. This is
@@ -175,6 +172,10 @@ func (s *ReleaseServer) performUpdateForce(req *services.UpdateReleaseRequest) (
 		}
 		return res, err
 	}
+
+	// update new release with next revision number so as to append to the old release's history
+	newRelease.Version = oldRelease.Version + 1
+	res.Release = newRelease
 
 	if req.DryRun {
 		s.Log("dry run for %s", newRelease.Name)

